extractor_service/main.py

```
@@ -0,0 +1,104 @@
+    elif ext == ".docx":
+        doc = docx.Document(file_path)
+        text = "\n".join([para.text for para in doc.paragraphs])
+    elif ext == ".txt":
+        with open(file_path, "r", encoding="utf-8") as f:
+            text = f.read()
+    else:
+        raise ValueError("Unsupported file type")
+    return text
+
+@app.post("/extract")
+async def extract_entities(file: UploadFile = File(...)):
```

Ruff (0.12.2)

Do not perform function call `File` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)